bin/doh.js

```diff
@@ -68,64 +38,75 @@ function dohLookup(options) {
+  description: 'DNS over HTTPS lookup command line tool',
+  prefixChars: '-+' // dig has done this to me
 });
-parser.addArgument(
-  'url',
-  {
-    help: 'URL to send the DNS request to'
-  }
-);
-parser.addArgument(
-  ['-m', '--method'],
-  {
-    help: 'Request method to use (GET or POST). Default is "POST"',
-    choices: ['GET', 'POST']
-  }
-);
-parser.addArgument(
-  ['-q', '--qname'],
-  {
-    help: 'Name to query for. Default is "."',
-    defaultValue: '.'
-  }
-);
-parser.addArgument(
-  ['-t', '--qtype'],
-  {
-    help: 'Query type. Default is "A"',
-    defaultValue: 'A'
-  }
-);
-parser.addArgument(
-  ['--ecs', '--subnet'],
-  {
-    help: 'EDNS Client Subnet option to include, in the format <address>/<source-prefix-len>',
-    dest: '_subnet',
-    metavar: '<address>/<source-prefix-len>'
-  }
-);
+
+// BASE POSITIONALS
+parser.addArgument('url',
+    {help: 'URL to send the DNS request to'});
+parser.addArgument('qname',
+    {help: 'Name to query for. Default is "."', defaultValue: '.', nargs: '?'});
+parser.addArgument('qtype',
+    {help: 'Query type. Default is "A"', defaultValue: 'A', nargs: '?'});
+
+// DNS OPTIONS (+)
+parser.addArgument('+subnet',
+  {help: 'EDNS Client Subnet option to include, in the format <address>/<source-prefix-len>', dest: '_subnet',
+    metavar: '<address>/<source-prefix-len>'});
+parser.addArgument('+dnssec',
+    {help: 'Send DNSSEC OK bit', action: 'storeTrue'});
+parser.addArgument('+edns',
+    {help: 'Send arbitrary EDNS options in the format of <opt-code>:<hex-data>. Can repeat arg multiple times',
+      metavar: '<opt-code>:<hex-data>', action: 'append', dest: '_edns'})
+
+// OTHER OPTIONS (-)
+parser.addArgument(['-m', '--method'],
+    {help: 'Request method to use (GET or POST). Default is "POST"', choices: ['GET', 'POST'], defaultValue: 'POST'});
+parser.addArgument(['-s', '--short'],
+    {help: 'Prints in short form (1 line) as opposed to pretty-printed json', action: 'storeTrue'});
+parser.addArgument(['-d', '--debug'],
+    {help: 'Prints the generated query in addition to the response', action: 'storeTrue'})
+
+
+// PARSING
 let args = parser.parseArgs();
 if (args._subnet) {
   let split = args._subnet.split('/');
   if (split.length !== 2) {
-    parser.exit(1, 'Invalid format for --ecs/--subnet option. Must be "<address>/<source-prefix-len>"\n');
+    parser.exit(1, 'Invalid format for +subnet option. Must be "<address>/<source-prefix-len>"\n');
   }
   args.ecsAddress = split[0];
   args.sourcePrefixLength = split[1];
 }
+if (args._edns) {
+  args.edns = []
+  for (const opt of args._edns) {
+    let split = opt.split(':')
+    if (split.length !== 2) {
+      parser.exit(1, 'Invalid format for +edns. Must be"<opt-code>:<hex-data>"\n')
+    }
+    args.edns.push({code: parseInt(split[0]), data: Buffer.of(split[1], 'hex')})
+  }
+}
 
-dohLookup(args)
-  .then(response => {
-    console.log(JSON.stringify(response))
+// LOOKUP
+const [query, response] = dohLookup(args);
```

Comment:
Might be cleaner if we put something like this here:
```javascript
delete args._edns;
delete args._subnet;
const [query, response] = dohLookup(args);
```
That way we don't have extra attributes hanging around